framework/src/media/utils/MediaUtils.cpp

```diff
@@ -54,6 +79,53 @@ void toUpperString(std::string &str)
 	}
 }
 
+audio_container_t getAudioContainerFromPath(std::string datapath)
+{
+	std::string basename = datapath.substr(datapath.find_last_of("/") + 1);
+	std::string extension;
+
+	if (basename.find(".") == std::string::npos) {
+		extension = "";
+	} else {
+		extension = basename.substr(basename.find_last_of(".") + 1);
+	}
+
+	toLowerString(extension);
+
+	if (extension.compare("wav") == 0) {
+		return AUDIO_CONTAINER_WAV;
+	} else if ((extension.compare("ogg") == 0) || (extension.compare("oga") == 0)) {
+		return AUDIO_CONTAINER_OGG;
+	} else if ((extension.compare("mp4") == 0) || (extension.compare("m4a") == 0)) {
+		return AUDIO_CONTAINER_MP4;
+	} else if (extension.compare("ts") == 0) {
+		return AUDIO_CONTAINER_MPEG2TS;
+	} else {
+		auto audioType = getAudioTypeFromPath(datapath);
```

Comment:
here we don't have to check audiotype too. we can check only container. Also we can use only one within NONE & UNKNOWN
```
audio_container_t container = getAudioContainerFromPath(path);
audio_type_t type;
switch (container) {
    case AUDIO_CONTAINER_WAV :
           type = wav;
    ....
    case AUDIO_CONTAINER_UNKNOWN:
           type = getAudioTypeFromPath(path);
           break;
     default :
           type = AUDIO_TYPE_INVALID;
           break;
 }
```